### PR TITLE
Psalm

### PR DESCRIPTION
**Pull request by JeremieSamson**
Add [psalm](https://github.com/vimeo/psalm) into the project and correct basic errors

### CODE REVIEW

File: src/DataFixtures/UniversityFixtures.php [hunk 1]
```diff
@@ -20,15 +20,16 @@ public function load(ObjectManager $manager)
         $universities = new ArrayCollection();
         $sectionId = -1;
         for ($i = 0; $i < self::NB_UNIVERSITY_TO_LOAD; ++$i) {
-            if (0 == substr($i, -1)) {
-                ++$sectionId;
-            }
+            //if (0 === substr($i, -1)) {
```
Comment (JeremieSamson) [inline src/DataFixtures/UniversityFixtures.php]: substr needs a string as first arg here, what did you wanted to do ?

File: src/DataFixtures/UniversityFixtures.php [hunk 2]
```diff
@@ -20,15 +20,16 @@ public function load(ObjectManager $manager)
...
+            //    ++$sectionId;
+            //}
             $university = new University();
-            $university->setName(self::UNIV_NAMES[substr($i, -1)]);
+            $university->setName(self::UNIV_NAMES[rand(0, \count(self::UNIV_NAMES))]);
```
Comment (JeremieSamson) [inline src/DataFixtures/UniversityFixtures.php]: same here, I changed to a random

File: src/Entity/User.php [hunk 3]
```diff
@@ -785,9 +785,9 @@ public function removeReport(Report $report): void
     {
         $this->reports->removeElement($report);
 
-        if ($report->getMentor() === $this) {
+        /*if ($report->getMentor() === $this) {
             $report->setMentor(null);
```
Comment (JeremieSamson) [inline src/Entity/User.php]: setMentor doesn't accept null here

File: psalm.xml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0"?>
+<psalm
+    totallyTyped="false"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns="https://getpsalm.org/schema/config"
+    xsi:schemaLocation="https://getpsalm.org/schema/config file:///Users/jeremie/Sites/buddysystem.eu/vendor/vimeo/psalm/config.xsd"
```
Comment (victor-paumier) [inline psalm.xml]: sounds bad

File: src/Controller/DashboardController.php [hunk 5]
```diff
@@ -88,7 +88,7 @@ public function reasonRefusalMatching(
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
-            $buddy->setConfirmed(false);
+            $buddy->setConfirmed(1);
```
Comment (victor-paumier) [inline src/Controller/DashboardController.php]: false == 1 ????

File: src/DataFixtures/UniversityFixtures.php (hunk 1 above)
Comment (victor-paumier) [reply src/DataFixtures/UniversityFixtures.php]: modulo 10 sounds better

File: src/DataFixtures/UniversityFixtures.php (hunk 2 above)
Comment (victor-paumier) [reply src/DataFixtures/UniversityFixtures.php]: random could give you multiple time same university name for a single section..

File: src/DataFixtures/UserFixtures.php [hunk 6]
```diff
@@ -32,30 +32,30 @@ public function load(ObjectManager $manager)
 
         // Super Admin
         $admin = new User();
-        $admin->setEnabled(1);
+        $admin->setEnabled(true);
```
Comment (victor-paumier) [inline src/DataFixtures/UserFixtures.php]: why did you do the invert in dashboardController?

File: src/Entity/Hobby.php [hunk 7]
```diff
@@ -59,12 +59,12 @@ public function getId(): int
         return $this->id;
     }
 
-    public function getName(): ?string
+    public function getName(): string
```
Comment (victor-paumier) [inline src/Entity/Hobby.php]: Have you try to create a hobby after this change?

File: src/Entity/Motivation.php [hunk 8]
```diff
@@ -54,12 +54,12 @@ public function getId(): int
         return $this->id;
     }
 
-    public function getName(): ?string
+    public function getName(): string
```
Comment (victor-paumier) [inline src/Entity/Motivation.php]: Have you try to create a motivation after this change?

File: src/Entity/Section.php [hunk 9]
```diff
@@ -469,7 +469,7 @@ public function setName($name): void
         $this->name = $name;
     }
 
-    public function getName(): ?string
+    public function getName(): string
```
Comment (victor-paumier) [inline src/Entity/Section.php]: Have you try to create?

File: src/Entity/University.php [hunk 10]
```diff
@@ -80,7 +80,7 @@ public function setSection(Section $section = null): void
         $this->section = $section;
     }
 
-    public function getSection(): ?Section
+    public function getSection(): Section
```
Comment (victor-paumier) [inline src/Entity/University.php]: Have you try to create?

File: src/Controller/DashboardController.php (hunk 5 above)
Comment (JeremieSamson) [reply src/Controller/DashboardController.php]: `@ORM\Column(type="integer", nullable=true)`
and
`public function setConfirmed(?int $confirmed): void`

File: src/DataFixtures/UserFixtures.php (hunk 6 above)
Comment (JeremieSamson) [reply src/DataFixtures/UserFixtures.php]: enabled is a boolean, not an integer

File: src/Entity/Hobby.php (hunk 7 above)
Comment (JeremieSamson) [reply src/Entity/Hobby.php]: did you ? :P

File: src/Entity/Motivation.php (hunk 8 above)
Comment (JeremieSamson) [reply src/Entity/Motivation.php]: did you ?

File: src/Entity/Section.php (hunk 9 above)
Comment (JeremieSamson) [reply src/Entity/Section.php]: did you ?

File: src/Entity/University.php (hunk 10 above)
Comment (JeremieSamson) [reply src/Entity/University.php]: did you ?

File: src/Entity/Hobby.php (hunk 7 above)
Comment (victor-paumier) [reply src/Entity/Hobby.php]: not working

File: src/Entity/Motivation.php (hunk 8 above)
Comment (victor-paumier) [reply src/Entity/Motivation.php]: not working

File: src/Entity/Section.php (hunk 9 above)
Comment (victor-paumier) [reply src/Entity/Section.php]: not working

**Pull request merged**